src/component/Restaurants.js

```diff
@@ -1,35 +1,57 @@
 import React, { useState } from "react";
+import PropTypes from "prop-types";
+import { RestaurantListItem } from "./RestaurantListItem";
 
-const api = "/cities";
+import thumbnail1Src from "place-my-order-assets/images/1-thumbnail.jpg";
+import thumbnail2Src from "place-my-order-assets/images/2-thumbnail.jpg";
+import thumbnail3Src from "place-my-order-assets/images/3-thumbnail.jpg";
+import thumbnail4Src from "place-my-order-assets/images/4-thumbnail.jpg";
 
 /**
  * Displays information about available restaurants.
  * @returns {JSX.Element}
  */
 function Restaurants (props) {
-    const [region, setRegion] = useState("");
     const [cities, setCities] = useState({});
-    const [city, setCity] = useState("");
+    const [city, setCity] = useState(DEFAULT_SELECT_VALUE);
+    const [region, setRegion] = useState(DEFAULT_SELECT_VALUE);
+    /** @type {[RestaurantRegionData, (value: RestaurantRegionData) => void]} */
+    const [restaurants, setRestaurants] = useState({});
 
+    function cityChange(evt) {
+        const { value: nextCityValue } = evt.target;
+        setCity(nextCityValue);
 
-    const cityChange = evt => {
-        setCity(evt.target.value);
-    };
+        if (nextCityValue && (!restaurants[region] || !restaurants[region][nextCityValue])) {
+            fetchRestaurants(region, nextCityValue);
+        }
+    }
 
-    const regionChange = evt => {
+    function regionChange(evt) {
         const { value: nextRegionValue } = evt.target;
         setRegion(nextRegionValue);
-        setCity("");
+        setCity(DEFAULT_SELECT_VALUE);
 
-        if (!cities[nextRegionValue]) {
+        if (nextRegionValue && !cities[nextRegionValue]) {
             fetchCities(nextRegionValue);
         }
-    };
+    }
 
-    const fetchCities = async cityRegion => {
-        const response = await fetch(`${api}?filter[state]=${cityRegion}`);
-        const body = await response.json();
+    async function fetchCities(cityRegion) {
+        const body = await apiFetch(`/cities?filter[state]=${cityRegion}`);
         setCities({ ...cities, [cityRegion]: body.data });
+    }
+
+    async function fetchRestaurants(filterRegion, filterCity) {
+        const body = await apiFetch(`/restaurants?filter[address.state]=${filterRegion}&filter[address.city]=${filterCity}`);
```

Comment:
We could pull this out into a "model" file, so the URL structure is not needed to be known by the react.

```js
// api.js
export async function getRestaurants({ ... }) {
  const response await fetch(`/restaurants...`);
  return response.json();
}
```